torch/csrc/jit/fusion_compiler.cpp

```diff
@@ -162,6 +209,31 @@ const char * scalarTypeName(at::ScalarType type) {
   }
 }
 
+std::string encodeRHS(Node * n) {
+  TemplateEnv env;
+  size_t i = 0;
+  for(auto in : n->inputs()) {
+    env.s(std::to_string(i++),nodeName(in));
+  }
+  // ops like div have a / b or a / 2 with the constant having the attribute other
+  // so we add other as an input if it is present
+  // 'pow' is the same but uses exponent as the attribute, so we handle that here as well
+  if(n->hasAttribute(kother) || n->hasAttribute(kexponent)) {
+    env.s(std::to_string(i), scalarValue(n->t(kother)));
+  }
```

Comment:
ugh, that's so ugly. I really feel that annotating the code snippets would be much nicer and clearer.

Reply:
Well, we currently lose the names of the inputs when we recordTrace (the tracing code doesn't save it), but if we somehow preserved it, you could explicitly name the parameters with self/other and then the special case here wouldn't be necessary, because the tensor input name and the attribute name coincide. But since we don't have that info... this seems reasonably clear to me.

Reply:
We are going to remove Scalars from ATen anyway, so I wouldn't worry too much about the particular workarounds we have to do now. Eventually `other` won't exist.